File: src/main/java/com/leijendary/spring/iamtemplate/generator/HtmlGenerator.java
package com.leijendary.spring.iamtemplate.generator;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;
import org.thymeleaf.context.Context;
import org.thymeleaf.spring5.SpringTemplateEngine;

import java.util.Map;

import static com.leijendary.spring.iamtemplate.util.RequestContext.getLocale;

@Component
@RequiredArgsConstructor
public class HtmlGenerator {

    private final SpringTemplateEngine templateEngine;

    public String parse(final String templateName, final Map<String, Object> parameters) {
        final var context = new Context(getLocale(), parameters);

        return templateEngine.process(templateName, context);
    }
}

File: src/main/kotlin/com/leijendary/spring/template/iam/generator/HtmlGenerator.kt
package com.leijendary.spring.template.iam.generator

import com.leijendary.spring.template.iam.core.util.RequestContext.locale
import org.springframework.stereotype.Component
import org.thymeleaf.context.Context
import org.thymeleaf.spring6.SpringTemplateEngine

@Component
class HtmlGenerator(private val templateEngine: SpringTemplateEngine) {
    fun parse(templateName: String, parameters: Map<String, Any?>): String {
        val context = Context(locale, parameters)

        return templateEngine.process(templateName, context)
    }
}
